src/pages/QuizUi/QuestionContent.jsx

```diff
@@ -0,0 +1,253 @@
+import React, { useState } from 'react';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+import { atomDark } from 'react-syntax-highlighter/dist/esm/styles/prism';
+import {
+  Container,
+  Typography,
+  RadioGroup,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  FormGroup,
+} from '@mui/material';
+import checkRadioRender from './CheckRadioRender';
+
+export default function QuestionContent({
+  question,
+  code,
+  options,
+  onAnswerSelected,
+  onSkipQuestion,
+  onBackQuestion,
+  index,
+  length,
+  answer,
+  type,
+  userAnswers,
+  questionId,
+  setFinishQuiz,
+}) {
+  const [selected, setSelected] = useState(type === `check-box` ? [] : '');
+  const [answeredQuestions, setAnsweredQuestions] = useState(new Set());
+  const [showWarningModal, setShowWarningModal] = useState(false);
+  const [showFeedbackCorrectModal, setShowFeedbackCorrectModal] =
+    useState(false);
+  const [showFeedbackIncorrectModal, setShowFeedbackIncorrectModal] =
+    useState(false);
+  const [showFinalQModal, setShowFinalQModal] = useState(false);
+
+  const resetSelection = () => {
+    setSelected('');
+  };
+
+  const isCurrentQuestionAnswered = answeredQuestions.has(questionId);
+
+  const handleNextQuestion = () => {
+    if (!selected && !isCurrentQuestionAnswered) {
+      setShowWarningModal(true);
+    } else {
+      onSkipQuestion(resetSelection);
+    }
+  };
+
+  const handleSubmitAnswer = () => {
+    const isCorrect = selected === answer.toString();
+    setAnsweredQuestions((prev) => new Set(prev).add(questionId));
+    if (isCorrect) {
+      setShowFeedbackCorrectModal(true);
+    } else {
+      setShowFeedbackIncorrectModal(true);
+    }
+  };
+
+  const handleSubmitTest = () => {
+    console.log(userAnswers, length);
+    const hasUnansweredQuestions =
+      userAnswers.length !== length || userAnswers.includes(undefined);
+    if (hasUnansweredQuestions) {
+      setShowFinalQModal(true);
+    } else {
+      setFinishQuiz(true);
+    }
+  };
+
+  return (
+    <Container>
+      <Typography variant="h5" align="left" sx={{ pl: 2, pb: 2 }}>
+        {question}
+      </Typography>
+      {code && (
+        <SyntaxHighlighter language="javascript" style={atomDark}>
+          {code}
+        </SyntaxHighlighter>
+      )}
+      {type === 'check-box' ? (
```

Comment:
The 'check-box' string is used a few times. In general if you use a string more than once it should be moved to a constant variable since it's easy to misspell strings. Our editors can't catch if we misspell strings since they can't know our intention. However a constant variable misspelling will be much easier to catch and our editors can do intellisense suggestions.

Reply:
Good suggestion!  Done!